catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/PenColorBrickTest.java

```diff
@@ -0,0 +1,210 @@
+import static org.catrobat.catroid.uiespresso.content.brick.utils.SeekbarsPickerWrapper.onColorSeekbar;
+
+@RunWith(Enclosed.class)
+public class PenColorBrickTest {
+
+	private static final Integer INIT_COLOR_VALUE = 128;
+
+	@RunWith(Parameterized.class)
+	public static class PenColorBrickTestParameterized {
+		@Rule
+		public BaseActivityInstrumentationRule<SpriteActivity> baseActivityTestRule = new
+				BaseActivityInstrumentationRule<>(SpriteActivity.class, SpriteActivity.EXTRA_FRAGMENT_POSITION, SpriteActivity.FRAGMENT_SCRIPTS);
+
+		@Before
+		public void setUp() throws Exception {
+			createProject();
+			baseActivityTestRule.launchActivity();
+		}
+
+		@Parameterized.Parameters(name = "{3}-Test")
+		public static Iterable<Object[]> data() {
+			return Arrays.asList(new Object[][] {
+					{R.id.color_rgb_seekbar_red,
+							R.id.brick_set_pen_color_action_red_edit_text,
+							"SeekbarRed",
+							(int) SeekbarsPickerWrapper.MAX_COLOR_SEEKBAR_VALUE},
+					{R.id.color_rgb_seekbar_green,
+							R.id.brick_set_pen_color_action_green_edit_text,
+							"SeekbarGreen",
+							(int) SeekbarsPickerWrapper.MIN_SEEKBAR_VALUE},
+					{R.id.color_rgb_seekbar_blue,
+							R.id.brick_set_pen_color_action_blue_edit_text,
+							"SeekbarBlue",
+							INIT_COLOR_VALUE},
+			});
+		}
+
+		@Parameterized.Parameter
+		public @IdRes int colorRgbSeekbarId;
+
+		@Parameterized.Parameter(1)
+		public @IdRes int brickActionEditTextId;
+
+		@Parameterized.Parameter(2)
+		public String testName;
+
+		@Parameterized.Parameter(3)
+		public int initValue;
+
+		@Test
+		public void testColorSeekBar() {
+			int whenBrickPosition = 0;
+			onBrickAtPosition(whenBrickPosition).checkShowsText(R.string.brick_when_started);
+			int penColorBrickPosition = 1;
+
+			onBrickAtPosition(penColorBrickPosition).onFormulaTextField(brickActionEditTextId)
+					.checkShowsNumber(initValue)
+					.perform(click());
+
+			onView(withId(colorRgbSeekbarId))
+					.perform(CustomSwipeAction.swipeLeftSlow());
+			onView(withId(colorRgbSeekbarId))
+					.perform(CustomSwipeAction.swipeRightSlow());
+			onColorSeekbar().pressBack(SeekbarsPickerWrapper.COLOR_SEEKBAR_MATCHER_RED);
+			onColorSeekbar().saveChanges();
+			onBrickAtPosition(penColorBrickPosition).onFormulaTextField(brickActionEditTextId)
+					.checkShowsNumber((int) SeekbarsPickerWrapper.MAX_COLOR_SEEKBAR_VALUE);
+		}
+	}
+
+	@RunWith(AndroidJUnit4.class)
+	public static class NotParameterizedPenColorBricksTest {
+
+		@Rule
+		public BaseActivityInstrumentationRule<SpriteActivity> baseActivityTestRule = new
+				BaseActivityInstrumentationRule<>(SpriteActivity.class, SpriteActivity.EXTRA_FRAGMENT_POSITION, SpriteActivity.FRAGMENT_SCRIPTS);
+
+		@Before
+		public void setUp() throws Exception {
+			createProject();
+			baseActivityTestRule.launchActivity();
+		}
+
+		@Test
+		public void testPenColorBrickUndoRedo() {
```

Comment:
please split this into smaller tests.
eg.

have a test that
- does one action on a seekbar and undo
- does two actions on a seekbar and undo
- does one action on one seekbar and one on another and undo
- etc....

try to keep tests as small as possible, do not try to test everything at once.
its always preferable to have 10 test methods with each 20 lines of code, instead of one with 100.